src/gleam/http.gleam

```diff
@@ -52,3 +61,163 @@ pub fn method_to_string(method) {
 
 pub external fn method_from_erlang(anything) -> Result(Method, Nil) =
   "gleam_http_native" "method_from_erlang"
+
+/// Data type containing an attribute value pair that constitute an HTTP header.
+pub type Header =
+  tuple(String, String)
+
+/// A general data type that can be be parameterised as a `Request` or `Response`
+pub opaque type Message(head, body) {
+  Message(head: head, headers: List(Header), body: body)
+}
+
+/// Data structure containing the deserialized components of an HTTP request start line.
+pub opaque type RequestHead {
```

Comment:
potentially these don't need to be opaque, but as it's a backwards compatible change to remove the opaque, but not to add it I decided to start this way.

Reply:
I think we should remove these and have a Request and Response type with no shared components. It's easier to understand and this API is likely the first one a user is to encounter, so I would like to stick to more familiar patterns unless there is an especially advantagerous reason to do otherwise.

Reply:
We seem to have gone full circle here. The reason for making them opaque was to have the shared components without the user needing to worry about it. There is a lot of shared functions basically everything that works on headers/body or both.

Reply:
I don't believe that there is any cost associated with having a set of functions for a request and a set for responses, and I can't find a single example outside of languages with classical inheritance that have this kind of abstraction around HTTP req/resps.